packages/toolkit/src/query/tests/buildHooks.test.tsx

```diff
@@ -730,6 +730,60 @@ describe('hooks tests', () => {
       expect(res.data!.amount).toBeGreaterThan(originalAmount)
     })
 
+    // See https://github.com/reduxjs/redux-toolkit/issues/4267 - Memory leak in useQuery rapid query arg changes
+    test('Hook subscriptions are properly cleaned up when query is fulfilled/rejected', async () => {
+      const pokemonApi = createApi({
+        baseQuery: fetchBaseQuery({ baseUrl: 'https://pokeapi.co/api/v2/' }),
+        endpoints: (builder) => ({
+          getTest: builder.query<string, number>({
+            async queryFn() {
+              await new Promise((resolve) => setTimeout(resolve, 1000));
+              return { data: "data!" };
+            },
+            keepUnusedDataFor: 0,
+          }),
+        }),
+      })
+
+      const storeRef = setupApiStore(pokemonApi, undefined, {
+        withoutTestLifecycles: true,
+      })
+
+      const checkNumQueries = (count: number) => {
+        const cacheEntries = Object.keys((storeRef.store.getState()).api.queries)
+        const queries = cacheEntries.length
+
+        expect(queries).toBe(count)
+      }
+
+      function User() {
+        const [fetchTest, { isFetching, isUninitialized }] =
+          pokemonApi.endpoints.getTest.useLazyQuery()
+
+        return (
+          <div>
+            <div data-testid="isUninitialized">{String(isUninitialized)}</div>
+            <div data-testid="isFetching">{String(isFetching)}</div>
+            <button data-testid="fetchButton" onClick={() => fetchTest(Math.random())}>
+              fetchUser
+            </button>
+          </div>
+        )
+      }
+
+      render(<User />, { wrapper: storeRef.wrapper })
+      fireEvent.click(screen.getByTestId('fetchButton'))
+      fireEvent.click(screen.getByTestId('fetchButton'))
+      fireEvent.click(screen.getByTestId('fetchButton'))
+
+      await act(async () => {
+        await delay(1000)
```

Comment:
As that hasn't been defined in this file yet, you'll have to define it somewhere at the top of the file

```ts
  function delay(ms: number) {
    return new Promise((resolve) => setTimeout(resolve, ms))
  }
```

Reply:
I believe you should also just be able to re-use the `delay` function from `msw`:

```ts
import { delay } from 'msw'

await delay(1000)
```

Reply:
My confusion here stems from the fact that it IS imported [here](https://github.com/reduxjs/redux-toolkit/blob/edafc36db93ca3dc6cfd8588f7e21fbb37e7aad2/packages/toolkit/src/query/tests/buildHooks.test.tsx#L39) and the test block directly below it uses it without problem.

Most likely I have just missed something painstakingly obvious, but I added a definition in the test block regardless.